src/NServiceBus.Transport.RabbitMQ.CommandLine/MigrateEndpointCommand.cs

```diff
@@ -0,0 +1,177 @@
+
+            if (routingTopology != Topology.Conventional)
+            {
+                throw new NotSupportedException("Quorum queue migration is only supported for the ConventionalRoutingTopology for the moment.");
+            }
+
+            CommandRunner.Run(connectionString, (connection, channel) =>
+            {
+                // make sure that the endpoint queue exists
+                channel.MessageCount(queueName);
+
+                //check if queue already is quorum
+                if (SafeExecute(connection, ch => ch.QueueDeclare(queueName, useDurableEntities, false, false, QuorumQueueArguments)))
+                {
+                    throw new Exception($"Queue {queueName} is already a quorum queue");
+                }
+
+                var holdingQueueName = $"{queueName}-migration-temp";
+
+                if (!SafeExecute(connection, ch =>
+                {
+                    if (ch.MessageCount(holdingQueueName) > 0)
+                    {
+                        throw new Exception($"Holding queue {holdingQueueName} has existing messages.");
+                    }
+                }))
+                {
+                    //does the holding queue need to be quorum?
+                    channel.QueueDeclare(holdingQueueName, true, false, false, QuorumQueueArguments);
+                    Console.WriteLine($"Holding queue created: {holdingQueueName}");
+                }
+
+                //bind the holding queue to the default exchange of queue under migration
+                // this will throw if the exchange for the endpoint doesn't exist
+                channel.QueueBind(holdingQueueName, queueName, EmptyRoutingKey);
+
+                Console.WriteLine($"Holding queue bound to main queue exchange");
+
+                //unbind the queue under migration to stopp more messages from coming in
+                channel.QueueUnbind(queueName, queueName, EmptyRoutingKey);
+                Console.WriteLine($"Main queue unbind");
+
+                //move all existing messages to the holding queue
+                var numMessagesMovedToHolding = ProcessMessages(
+                    channel,
+                    queueName,
+                    (message, channel) =>
+                    channel.BasicPublish(EmptyRoutingKey, holdingQueueName, message.BasicProperties, message.Body),
+                    cancellationToken);
+
+                Console.WriteLine($"{numMessagesMovedToHolding} messages moved to the holding queue");
+
+                // delete the queue under migration
+                channel.QueueDelete(queueName);
+                Console.WriteLine($"Main queue removed");
+
+                //recreate the queue
+                channel.QueueDeclare(queueName, useDurableEntities, false, false, QuorumQueueArguments);
+                Console.WriteLine($"Main queue recreated as a quorum queue");
+
+                channel.QueueBind(queueName, queueName, EmptyRoutingKey);
```

Comment:
Would it be better to wait to re-bind the new queue until after the messages have been migrated from the temp queue?  It would better preserve some of the original message order I think.